x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/SecurityIndexRolesMetadataMigrationIT.java

```diff
@@ -108,21 +134,49 @@ private void addExpectWarningOption(RequestOptions.Builder options) {
     }
 
     @SuppressWarnings("unchecked")
-    private void waitForMigrationCompletion() throws Exception {
-        final Request request = new Request("GET", "_cluster/state/metadata/.security-7");
+    private static void assertNoMigration(RestClient adminClient) throws Exception {
+        Request request = new Request("GET", "_cluster/state/metadata/" + INTERNAL_SECURITY_MAIN_INDEX_7);
+        Response response = adminClient.performRequest(request);
+        assertOK(response);
+        Map<String, Object> responseMap = responseAsMap(response);
+        Map<String, Object> indicesMetadataMap = (Map<String, Object>) ((Map<String, Object>) responseMap.get("metadata")).get("indices");
+        assertTrue(indicesMetadataMap.containsKey(INTERNAL_SECURITY_MAIN_INDEX_7));
+        assertFalse(
+            ((Map<String, Object>) indicesMetadataMap.get(INTERNAL_SECURITY_MAIN_INDEX_7)).containsKey(MIGRATION_VERSION_CUSTOM_KEY)
+        );
+    }
+
+    @SuppressWarnings("unchecked")
+    private static void waitForMigrationCompletion(RestClient adminClient, @Nullable Integer migrationVersion) throws Exception {
+        final Request request = new Request("GET", "_cluster/state/metadata/" + INTERNAL_SECURITY_MAIN_INDEX_7);
         assertBusy(() -> {
-            Response response = adminClient().performRequest(request);
+            Response response = adminClient.performRequest(request);
             assertOK(response);
             Map<String, Object> responseMap = responseAsMap(response);
+            Map<String, Object> indicesMetadataMap = (Map<String, Object>) ((Map<String, Object>) responseMap.get("metadata")).get(
+                "indices"
+            );
+            assertTrue(indicesMetadataMap.containsKey(INTERNAL_SECURITY_MAIN_INDEX_7));
             assertTrue(
-                ((Map<String, Object>) ((Map<String, Object>) ((Map<String, Object>) responseMap.get("metadata")).get("indices")).get(
-                    ".security-7"
-                )).containsKey("migration_version")
+                ((Map<String, Object>) indicesMetadataMap.get(INTERNAL_SECURITY_MAIN_INDEX_7)).containsKey(MIGRATION_VERSION_CUSTOM_KEY)
             );
+            if (migrationVersion != null) {
```

Comment:
`migrationVersion` is always null so I think this can be removed.

Reply:
I'll do that. This is what's left of a past attempt to make the `waitForMigrationCompletion` cmd reusable, but couldn't find a place for it without changing the modules deps.

Reply:
I see, that makes sense.

Reply:
Pushed [12648cc](https://github.com/elastic/elasticsearch/pull/108733/commits/12648cc76fb1f386a0cb75e619ff395662a47be8)